### DISCUSSION TITLE
Show all open buffers regardless of the "working directory"

### DISCUSSION THREAD

**Discussion by vakorol** [nvim-neo-tree/neo-tree.nvim#1407] [General]
Hi!
Thanks for the plugin, it looks very nice and so far proved to be the most convenient of all the alternatives!

One thing that is bothering me though - I would like to see the list of _all open buffers_, not the buffers for _files in the "current directory"_. The problem is - if I open Neovim in `~/repo/project1`, then `neo-tree` will happily show all open buffers under the directory. If at some point I do `:e ../project2/file.txt`, then opening `neo-tree` won't even show that file, since it will still be listing the files under `~/repo/project1`.

If I run with `dir=/`, to include all possible directories in the tree (because I can run `nvim` anywhere in the filesystem), then it works fine _for a while_, but at some point typing `:e <TAB>` starts showing the root of the filesystem, which is not what I'd expect it to do. I wrote "_for a while_" because the behavior is inconsistent and for some time `:e <TAB>` correctly shows the "current directory", but then it all of a sudden switches to `/`.

Can someone give a hint how to fix this? It would be great to have a setup where the tree lists all open buffers, nicely grouped by their parent directories, regardless of the "current project directory" but with `:e` pointing to it?

Thanks!

**Comment by vakorol** [comment]
Basically, i'm looking for an option for `neo-tree` to show all open buffers under _the topmost common parent directory_.

**Comment by pysan3** [comment]
Are you talking about the filesystem source or the buffers source?

**Reply by vakorol** [reply]
The buffers source, sorry for the confusion.

**Comment by pysan3** [comment]
You have `bind_to_cwd = false` for both filesystem and buffers source right?

**Reply by vakorol** [reply]
yes:

```lua
    {
        "nvim-neo-tree/neo-tree.nvim",
        branch = "v3.x",
        dependencies = {
          "nvim-lua/plenary.nvim",
          "nvim-tree/nvim-web-devicons", -- not strictly required, but recommended
          "MunifTanjim/nui.nvim",
          -- "3rd/image.nvim", -- Optional image support in preview window: See `# Preview Mode` for more information
        },
        config = function ()
            require('neo-tree').setup {
                buffers = {
                    leave_dirs_open = true,
                    follow_current_file = {
                        enabled = true,
                        leave_dirs_open = true,
                    },
                    bind_to_cwd = false,
                },
                filesystem = {
                    bind_to_cwd = false,
                },
            }
        end
    },
```

**Comment by pysan3** [comment]
Hmm I really can't reproduce.

Could you add this to your init.lua and live for a while?
It should report when cwd changes, and please tell me what you did at that right moment.

```lua
vim.api.nvim_create_autocmd("DirChanged", {
  callback = function(...)
    vim.print(string.format([[Dir Changed (%s): %s]], vim.fn.getcwd(), vim.inspect({ ... })))
  end,
})
```

**Reply by pysan3** [reply]
Also check that neo-tree is updated.

**Reply by vakorol** [reply]
Thanks! Will try monitor this.

**Comment by krishna-bala** [comment]
EDIT: I found some more discussions (https://github.com/nvim-neo-tree/neo-tree.nvim/issues/885, https://github.com/nvim-neo-tree/neo-tree.nvim/issues/1411). Seems like the solution is to use `:Neotree buffers dir=/` and that this is the intended behavior.

I am also curious if there is a way to show _all_ open buffers, not just the buffers open in the current project. 

I am using LazyVim. The setup in LazyVim is given at the bottom of this post for reference.

The issue I am experiencing is as follows:
When I run `:Neotree buffers`, it opens it with the title _OPEN BUFFERS in \<root path\>_. If I have buffers open in different project, it will not show those buffers that are not in the current project path. For example, let's say I have git projects `~/project1` and `~/project2` and I open `bufferA` from `project1` and `bufferB` from `project2`. If my current window is of `bufferA` and I run `:Neotree buffers`, it will open _Neo-Tree Buffers_ with the title _OPEN BUFFERS in ~/project1_. However, if I run `:ls` or `:buffers`, the output will contain both `bufferA` and `bufferB`. I can resolve this by running `:Neotree buffers dir=/`, as @vakorol mentioned (I am **not** having issues with my "working directory" changing after a while, most likely because of some nice features of LazyVim).

I'm not sure if this behavior is caused by the other plugins/configurations in LazyVim, so I am curious if this is expected behavior. I would match rather the output just show _OPEN BUFFERS_ and not have it be specific to the current root/project/cwd, but that's just me.

```lua
{
  "nvim-neo-tree/neo-tree.nvim",
  branch = "v3.x",
  cmd = "Neotree",
  keys = {
    {
      "<leader>fe",
      function()
        require("neo-tree.command").execute({ toggle = true, dir = LazyVim.root() })
      end,
      desc = "Explorer NeoTree (Root Dir)",
    },
    {
      "<leader>fE",
      function()
        require("neo-tree.command").execute({ toggle = true, dir = vim.uv.cwd() })
      end,
      desc = "Explorer NeoTree (cwd)",
    },
    { "<leader>e", "<leader>fe", desc = "Explorer NeoTree (Root Dir)", remap = true },
    { "<leader>E", "<leader>fE", desc = "Explorer NeoTree (cwd)", remap = true },
    {
      "<leader>ge",
      function()
        require("neo-tree.command").execute({ source = "git_status", toggle = true })
      end,
      desc = "Git Explorer",
    },
    {
      "<leader>be",
      function()
        require("neo-tree.command").execute({ source = "buffers", toggle = true })
      end,
      desc = "Buffer Explorer",
    },
  },
  deactivate = function()
    vim.cmd([[Neotree close]])
  end,
  init = function()
    -- FIX: use `autocmd` for lazy-loading neo-tree instead of directly requiring it,
    -- because `cwd` is not set up properly.
    vim.api.nvim_create_autocmd("BufEnter", {
      group = vim.api.nvim_create_augroup("Neotree_start_directory", { clear = true }),
      desc = "Start Neo-tree with directory",
      once = true,
      callback = function()
        if package.loaded["neo-tree"] then
          return
        else
          local stats = vim.uv.fs_stat(vim.fn.argv(0))
          if stats and stats.type == "directory" then
            require("neo-tree")
          end
        end
      end,
    })
  end,
  opts = {
    sources = { "filesystem", "buffers", "git_status", "document_symbols" },
    open_files_do_not_replace_types = { "terminal", "Trouble", "trouble", "qf", "Outline" },
    filesystem = {
      bind_to_cwd = false,
      follow_current_file = { enabled = true },
      use_libuv_file_watcher = true,
    },
    window = {
      mappings = {
        ["l"] = "open",
        ["h"] = "close_node",
        ["<space>"] = "none",
        ["Y"] = {
          function(state)
            local node = state.tree:get_node()
            local path = node:get_id()
            vim.fn.setreg("+", path, "c")
          end,
          desc = "Copy Path to Clipboard",
        },
        ["O"] = {
          function(state)
            require("lazy.util").open(state.tree:get_node().path, { system = true })
          end,
          desc = "Open with System Application",
        },
        ["P"] = { "toggle_preview", config = { use_float = false } },
      },
    },
    default_component_configs = {
      indent = {
        with_expanders = true, -- if nil and file nesting is enabled, will enable expanders
        expander_collapsed = "",
        expander_expanded = "",
        expander_highlight = "NeoTreeExpander",
      },
      git_status = {
        symbols = {
          unstaged = "󰄱",
          staged = "󰱒",
        },
      },
    },
  },
  config = function(_, opts)
    local function on_move(data)
      LazyVim.lsp.on_rename(data.source, data.destination)
    end

    local events = require("neo-tree.events")
    opts.event_handlers = opts.event_handlers or {}
    vim.list_extend(opts.event_handlers, {
      { event = events.FILE_MOVED, handler = on_move },
      { event = events.FILE_RENAMED, handler = on_move },
    })
    require("neo-tree").setup(opts)
    vim.api.nvim_create_autocmd("TermClose", {
      pattern = "*lazygit",
      callback = function()
        if package.loaded["neo-tree.sources.git_status"] then
          require("neo-tree.sources.git_status").refresh()
        end
      end,
    })
  end,
}
```
